Cannot find module 'https://cdn.jsdelivr.net/pyodide/dev/full/pyodide.asm.js'

Hi all. I have the following problem: I'm trying to use pyodide in a react. I installed the package using npm install, and imported the loadPyodide function from it. I have this code:
```
const pyodide = await loadPyodide({
             indexURL: "https://cdn.jsdelivr.net/pyodide/v0.24.0/full/",
             fullStdLib: true,
             stdin: window.prompt,
             stdout: (text) => {
                 if (text !== "Python initialization complete") {
                     setResult(prevState => prevState + text + "\n")
                 }
             },
             stderr: (text) => {
                 setResult(prevState => prevState + text + "\n")
             }
         });
```
However I get an error:
```
Cannot find module 'https://cdn.jsdelivr.net/pyodide/dev/full/pyodide.asm.js'
     at http://localhost:3000/static/js/bundle.js:277779:11
     at async loadScript (http://localhost:3000/static/js/bundle.js:317327:43)
     at async loadPyodide (http://localhost:3000/static/js/bundle.js:317657:5)
     at async loadPython (http://localhost:3000/static/js/bundle.js:41685:21)
     at async run (http://localhost:3000/static/js/bundle.js:41761:24)
```
How can you solve it? I would appreciate your answer.

I have the same issue when I try to upgrade to v0.24.*, using indexURL in the same manner:

https://github.com/tylfin/restructedpyjs/blob/186bf34ba5212c74d0825799a6ce156e58fdfc3a/src/restructedpyjs.js#L19-L24

The error stack I'm getting looks like:

```
Uncaught (in promise) Error: Cannot find module 'https://cdn.jsdelivr.net/pyodide/v0.24.1/full/pyodide.asm.js'
    ./node_modules/pyodide lazy recursive/t/< lazy strict namespace object:5
    promise callback*t lazy strict namespace object:4
    ...
```

Hmm, removing the option gives the same module error:

```
Uncaught (in promise) Error: Cannot find module 'http://localhost:8000/node_modules/pyodide/pyodide.asm.js'
    ./node_modules/pyodide lazy recursive/t/< lazy strict namespace object:5
    promise callback*t lazy strict namespace object:4
```

--- a/.gitignore
+++ b/.gitignore
@@ -17,6 +17,7 @@ __pycache__
 .pytest_cache/
 .vscode
 .venv*
+tsconfig.tsbuildinfo
 
 build
 ccache

--- a/Makefile
+++ b/Makefile
@@ -133,7 +133,7 @@ node_modules/.installed : src/js/package.json src/js/package-lock.json
 	touch node_modules/.installed
 
 dist/pyodide.js src/js/generated/_pyodide.out.js: src/js/*.ts src/js/generated/pyproxy.ts node_modules/.installed
-	cd src/js && npm run tsc && node esbuild.config.mjs && cd -
+	cd src/js && npm run build && cd -
 
 src/core/stack_switching/stack_switching.out.js : src/core/stack_switching/*.mjs
 	node src/core/stack_switching/esbuild.config.mjs

--- a/docs/project/changelog.md
+++ b/docs/project/changelog.md
@@ -23,6 +23,9 @@ myst:
 - {{ Fix }} Fixed default indexURL calculation in Node.js environment.
   {pr}`4288`
 
+- {{ Fix }} Fixed a bug that webpack messes up dynamic import of `pyodide.asm.js`.
+  {pr}`4294`
+
 - {{ Enhancement }} Added experimental support for stack switching.
   {pr}`3957`, {pr}`3964`, {pr}`3987`, {pr}`3990`, {pr}`3210`
 

--- a/src/js/esbuild.config.mjs
+++ b/src/js/esbuild.config.mjs
@@ -96,7 +96,14 @@ try {
     );
 
     // inject webpackIgnore comments
+<<<<<<< Updated upstream
     content = content.replace("import(", "import(/* webpackIgnore */");
+=======
+    content = content.replaceAll(
+      "import(",
+      "import(/* webpackIgnore */",
+    );
+>>>>>>> Stashed changes
 
     writeFileSync(outfile, content);
   }

--- a/src/js/esbuild.config.mjs
+++ b/src/js/esbuild.config.mjs
@@ -95,6 +95,12 @@ try {
       `try{Object.assign(exports,${name})}catch(_){}\nglobalThis.${name}=${name}.${name};\n//# sourceMappingURL`,
     );
 
+    // inject webpackIgnore comments
+    content = content.replace(
+      "import(",
+      "import(/* webpackIgnore */",
+    );
+
     writeFileSync(outfile, content);
   }
 } catch ({ message }) {

--- a/src/js/esbuild.config.mjs
+++ b/src/js/esbuild.config.mjs
@@ -96,10 +96,7 @@ try {
     );
 
     // inject webpackIgnore comments
-    content = content.replace(
-      "import(",
-      "import(/* webpackIgnore */",
-    );
+    content = content.replace("import(", "import(/* webpackIgnore */");
 
     writeFileSync(outfile, content);
   }

--- a/src/js/esbuild.config.mjs
+++ b/src/js/esbuild.config.mjs
@@ -96,14 +96,7 @@ try {
     );
 
     // inject webpackIgnore comments
-<<<<<<< Updated upstream
-    content = content.replace("import(", "import(/* webpackIgnore */");
-=======
-    content = content.replaceAll(
-      "import(",
-      "import(/* webpackIgnore */",
-    );
->>>>>>> Stashed changes
+    content = content.replaceAll("import(", "import(/* webpackIgnore */");
 
     writeFileSync(outfile, content);
   }

--- a/src/js/package.json
+++ b/src/js/package.json
@@ -79,6 +79,7 @@
     "ws": false
   },
   "scripts": {
+    "build": "tsc --noEmit && node esbuild.config.mjs",
     "test": "npm-run-all test:*",
     "test:unit": "cross-env TEST_NODE=1 ts-mocha --node-option=experimental-loader=./test/loader.mjs --node-option=experimental-wasm-stack-switching -p tsconfig.test.json test/unit/**/*.test.*",
     "test:node": "cross-env TEST_NODE=1 mocha test/integration/**/*.test.js",

--- a/src/js/tsconfig.json
+++ b/src/js/tsconfig.json
@@ -8,6 +8,7 @@
     "module": "ES2022",
     "moduleResolution": "node",
     "composite": true,
+    "tsBuildInfoFile": "tsconfig.tsbuildinfo",
     "strict": true,
     "noUnusedLocals": false,
     "types": ["node"],

